lib/crypto/client.cpp

```diff
@@ -507,6 +507,40 @@ SAS::calculate_mac(std::string input_data, std::string info)
         return to_string(output_buffer);
 }
 
+//! checks if the signature is signed by the signing_key
+bool
+OlmClient::ed25519_verify_sig(std::string signing_key, nlohmann::json obj, std::string signature)
+{
+        using namespace client::utils;
+
+        try {
+                if (signature.empty())
+                        return false;
+
+                obj.erase("unsigned");
+                obj.erase("signatures");
+
+                auto utility = create_olm_object<UtilityObject>();
+                auto ret     = olm_ed25519_verify(utility.get(),
+                                              signing_key.data(),
+                                              signing_key.size(),
+                                              obj.dump().data(),
```

Comment:
You should probably use a temporary variable for this to not serialize the same event twice.